[IMP] fieldservice_stock: Add "Create FSM Equipment" option to Operation Types

dreispt opened the pull request:
FSM Equipment creation for serial number tracked products can be
automated on outgoing/delivey transfers.
This allows for it to happen earlier, on other Operation Types.

dreispt committed:
[IMP] Add "Create FSM Equipment" option to Operation Types

FSM Equipment creation for serial number tracked products can be
on specific Operation Types, such as Delivery or Receipts.

max3903 commented on fieldservice_stock/models/stock_move.py:
```diff
@@ -20,8 +20,11 @@ def prepare_equipment_values(self, move_line):
+            if rec.state == 'done' and create_fsm_equipment:
+                is_outgoing = rec.picking_code == 'outgoing'
+                can_create_fsm = rec.picking_type_id.can_create_fsm_equipment
+                if can_create_fsm or is_outgoing:
```
@dreispt 

I think we can simplify this by dropping the picking_code. I can easily see a company  willing to create an equipment when a product is received.

```
if rec.state == 'done' and \
    rec.product_tmpl_id.create_fsm_equipment and \
    rec.picking_type_id.create_fsm_equipment:
```

dreispt replied:
I worry about being "stable" and not breaking current installs.
Do you think that is something we could waive?

max3903 replied:
@dreispt We can have a migration script that check the box on the picking type based on the code.

max3903 commented on fieldservice_stock/models/stock_picking_type.py:
```diff
@@ -0,0 +1,16 @@
+class StockPickingType(models.Model):
+    _inherit = 'stock.picking.type'
+
+    force_create_fsm_equipment = fields.Boolean(
```
```suggestion
    create_fsm_equipment = fields.Boolean(
```

dreispt committed:
[IMP] Migration script to handle the otherwise breaking changes

dreispt commented:
@max3903 Doing tests, I stumbled upon a required field:
`ERROR: null value in column "current_location_id" violates not-null constraint`
Any suggestion on how to deal with this?

dreispt committed:
[FIX] FSM Equipment: current location required on View only

Some workflows need to create an FSM Equipemnt that is not yet assigned
to a location.
For example, FSM Equipments still in company warehouse waiting to be
delivered to service locations.

dreispt commented:
Ready for review.

max3903 committed:
Merge branch '12.0' into 12.0-imp-force-create-fsm

dreispt commented:
@max3903 Good to merge?

dreispt commented:
/ocabot merge

dreispt committed:
Merge branch '12.0' into 12.0-imp-force-create-fsm

dreispt commented:
/ocabot merge minor
